src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -1260,22 +1260,28 @@ public boolean isSatisfiedBy(TableMetadata metadata, DecoratedKey partitionKey,
         private boolean contains(TableMetadata metadata, DecoratedKey partitionKey, Row row)
         {
             assert column.type.isCollection();
-            CollectionType<?> type = (CollectionType<?>)column.type;
+            assert (indexAnalyzer == null) == (queryAnalyzer == null);
```

Comment:
This is the assertion test coverage complains about

Reply:
In my opinion, the point of assertions is to test invariants that are not currently reachable but might become reachable in the future but that would break this code.

Reply:
> In my opinion, the point of assertions is to test invariants that are not currently reachable but might become reachable in the future but that would break this code.

Exactly, as mentioned in Slack I marked them mostly for history. 
If later someone wants to check the coverage for an old PR - they see only the percentage and they cannot get to the details in Sonar anymore.